### ISSUE TITLE
Adding new product from Purchase page redirects to wrong URL causing HTTP 404

### ISSUE THREAD

**Issue by LordBoos** [hassio-addons/addon-grocy#36]
# Problem/Motivation

>  When I add a new product from the Purchase page, Page Not Foud page is shown, because the URL in redirect looks wrong.
This works and it is correct URL: https://XXX.duckdns.org/api/hassio_ingress/XXX/api/objects/products
But after adding the product and clicking Save, this URL is called: https://XX.duckdns.org/api/hassio_ingress/XXX/https://XXX.duckdns.orgpurchase?createdproduct=Mra%C5%BEen%C3%A1%20zelenina%20-%20pol%C3%A9vkov%C3%A1. It returns 404 and Grocy shows "Page Not Found" page.

## Expected behavior

> I expect to be returned to previous purchase page after adding new product on purchase page.

## Actual behavior

> Wrong URL is called, it returns 404 and Grocy shows "Page Not Found" page.

## Steps to reproduce

> 

1. Open Grocy using Hassio Ingress
1. Go to Purchase page
1. Write some new Name to Product field and press TAB
1. Select Add as new product
1. Fill in the form and press Save

## Proposed changes

> No

## Additional information

> This happens only using Hassio Ingress. Using direct access via port works fine.

**Comment by sinclairpaul** [comment]
Possible duplicate of #35 can you share your config and confirm if the same issue occurs if the addon is exposed on a port rather than using ingress?

**Comment by Boosik** [comment]
Addon config:
```
{
  "culture": "en",
  "currency": "CZK",
  "features": {
    "batteries": true,
    "calendar": true,
    "chores": true,
    "equipment": true,
    "recipes": true,
    "shoppinglist": true,
    "stock": true,
    "tasks": true
  },
  "ssl": true,
  "certfile": "fullchain.pem",
  "keyfile": "privkey.pem",
  "log_level": "info"
}
```
I|ve stated in Additional information that this happens only using Ingress. Exposing on a port and accessing it directly works fine without this issue. Also it is not related to Currency as is stated in #35 because it was happening for me even with default USD currency. I changed it to CZK only after I faced this issue for the first time.

**Comment by sinclairpaul** [comment]
Sorry I missed the comment in Additional Info, will look into the Ingress issue, for the time being keep the addon exposed.

**Comment by sinclairpaul** [comment]
Looks to be to do with the Javascript here https://github.com/grocy/grocy/blob/b76e51ba41236b30527df076debd04fff0171405/public/js/extensions.js#L24

@frenck Not sure we can fix this

**Comment by Boosik** [comment]
I hope it can be fixed somehow, because this is pretty much breaking issue for ingress and I'm not comfortable with having Grocy exposed without 2FA on public url:port for a long time 😢

**Comment by frenck** [comment]
We should find a way to report this in with Grocy.

Grocy supports reverse proxy/hosting in subfolders, however, in this case, it goes south.

**Comment by sinclairpaul** [comment]
The issue isn't a sub folder though, its a proxy to another proxy, the Javascript appears to look for the search string for returnto like:
http://192.168.1.92:9222/product/new?prefillname=adsadsafefe&returnto=%2Fpurchase

Where under ingress all browser data will be seen as:

http://192.168.1.92:8123/hassio/ingress/local_grocy

So the fix would need to be change to server side rather than client (which is beyond my limited expertise :) ).

I will add at this point the issue is cosmetic, i.e. any posts will complete, it is just the return URL that fails.

**Comment by Boosik** [comment]
@sinclairpaul it is cosmetic but sometimes it's blocking issue.
Eg. I use Grocy on a tablet fixed on my fridge and when this issue happen I'm locked out on 404 page and can't return back because the "Go back to home screen link" doesn't work either. So only way to fix it is to force close "Grocy app" (in fact it's chrome shortcut, but it behaves like an app) and open it again because there is no access to address bar or HA sidebar.

**Comment by sinclairpaul** [comment]
In all honesty I'm not sure we can fix this, its an issue that is caused by the way ingress works.  All I can suggest at this point is access the addon via the port from your fridge, and use ingress externally, it should work.

**Comment by ocewers** [comment]
What is the proposed workaround at the moment for this problem?
